OSDOCS#7588:Adds a batch of bug fixes to MicroShift RNs (#65976)
openshift/openshift-docs: microshift_release_notes/microshift-4-14-release-notes.adoc

* Previously, certificate authorities generated `kubeconfig` files for {product-title}'s embedded components. With this update, certificate authorities are reconfigured to ensure the `kubeconfig` files are independent. (link:https://issues.redhat.com/browse/OCPBUGS-8301[*OCPBUGS-8301*])

> * Previously, the sos wrapper helper was not present and the sos command manually. With this update, the sos wrapper was added the {microshift-short} and you no longer need to run the sos command manually when debugging. (link:https://issues.redhat.com/browse/OCPBUGS-19772[*OCPBUGS-19772*])

Reviewer: ```suggestion
* Previously, the sos wrapper helper was not present and the sos command had to be run manually. With this update, the sos wrapper is added to {microshift-short} and you do not need to run the sos command manually when debugging. (link:https://issues.redhat.com/browse/OCPBUGS-19772[*OCPBUGS-19772*])
```

> * Previously, {microshift-short} did not generate version details at startup. With this update, {microshift-short} now logs version details at startup. (link:https://issues.redhat.com/browse/OCPBUGS-19638[*OCPBUGS19638*])

Reviewer: ```suggestion
* Previously, {microshift-short} did not generate version details at startup. With this update, {microshift-short} now logs version details at startup. (link:https://issues.redhat.com/browse/OCPBUGS-19638[*OCPBUGS-19638*])
```

> * Previously, node IP certificates were using the old IP instead of the new causing all external certificates not to verify against the new IP. With this update, certificates update when the node IP changes. (link:https://issues.redhat.com/browse/OCPBUGS-19315[*OCPBUGS-19315*])

Reviewer: ```suggestion
* Previously, node IP certificates continued to use the previous IP address when the IP address changed, causing external certificates not to verify against the new IP address. With this update, certificates update when the node IP changes. (link:https://issues.redhat.com/browse/OCPBUGS-19315[*OCPBUGS-19315*])
```

> * Previously, changing IP results left `kubeconfigs` for the old IP on disk making the `kubeconfigs` unusable because they target an IP that does not belong to {microshift-short} anymore. With this update, the PLD `kubeconfigs` are deleted when the IP changes. (link:https://issues.redhat.com/browse/OCPBUGS-15740[*OCPBUGS-15740*])

Reviewer: ```suggestion
* Previously, after changing the IP address, `kubeconfig` files for the previous IP remained. These `kubeconfig` files were unusable because they targeted an IP not used by {microshift-short}. With this update, the `kubeconfig` files associated with the previous IP address are deleted when the IP changes. (link:https://issues.redhat.com/browse/OCPBUGS-15740[*OCPBUGS-15740*])
```

> * Previously, KCM and KAS did not exit even after getting a cancellation signal. With this update, {microshift-short} shuts down as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18773[*OCPBUGS-18773*])

Reviewer: ```suggestion
* Previously, KCM and KAS did not exit after getting a cancellation signal. With this update, {microshift-short} shuts down as expected. (link:https://issues.redhat.com/browse/OCPBUGS-18773[*OCPBUGS-18773*])
```

> * Previously, when stopping `microshift.service`, `microshift-etcd.scope` is also signaled to stop which caused etcd to stop too soon. With this update, the issue that prevented the etcd database from shutting down cleanly is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-14678[*OCPBUGS-14678*])

Reviewer: ```suggestion
* Previously, when stopping `microshift.service`, `microshift-etcd.scope` was also signaled to stop which caused etcd to stop too soon. With this update, the issue that prevented the etcd database from shutting down in the correct order is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-14678[*OCPBUGS-14678*])
```

> * Previously, the `microshift show-config` command did not show the configuration changes. With this update the behavior of the `microshift show-config` command shows the effective configuration by default. (link: https://issues.redhat.com/browse/OCPBUGS-13221[*OCPBUGS-13221*])

Reviewer: ```suggestion
* Previously, the `microshift show-config` command did not show configuration changes. With this update, the `microshift show-config` command shows the current configuration by default. (link: https://issues.redhat.com/browse/OCPBUGS-13221[*OCPBUGS-13221*])
```

> * Previously, the `externalGatewayInterface` flag was used to specify the interface to be added to the OVS gateway bridge `br-exl` to be used as a second gateway interface. When {microshift-short} switched to OVN-K local gateway mode, all pods to external traffic goes through the host kernal before entering or leaving the host. With this update, the unused `externalGatewayInterface` configuration parameter from the `ovn.yaml` file for the CNI driver was removed. (link:https://issues.redhat.com/browse/OCPBUGS-13078[*OCPBUGS-13078*])

Reviewer: ```suggestion
* Previously, the `externalGatewayInterface` flag was used to specify the interface to be added to the OVS gateway bridge `br-exl` to be used as a second gateway interface. {microshift-short} switched to the OVN-K local gateway mode. All pods-to-external traffic now goes through the host kernal before entering or leaving the host. With this update, the unused `externalGatewayInterface` configuration parameter from the `ovn.yaml` file for the CNI driver is removed. (link:https://issues.redhat.com/browse/OCPBUGS-13078[*OCPBUGS-13078*])
```

> * Previously, the search logic for directories with kustomization files searched for `kustomization.yaml`. `kubectl` also searches for `kustomization.yaml` and `kustomization`. With this update, the search logic is extended for manifest input files to include `kustomization.yml` and `kustomization` in addition to the existing `kustomization.yaml`. (link:https://issues.redhat.com/browse/OCPBUGS-12744[*OCPBUGS-12744*])

Reviewer: ```suggestion
* Previously, the search logic for directories with `kustomization` files searched for `kustomization.yaml`. `kubectl` also searches for `kustomization.yaml` and `kustomization`. With this update, the search logic is extended for manifest input files to include `kustomization.yml` and `kustomization` in addition to the existing `kustomization.yaml`. (link:https://issues.redhat.com/browse/OCPBUGS-12744[*OCPBUGS-12744*])
```
